docs/architecture/architecture.md

```diff
@@ -0,0 +1,378 @@
+# Clear Containers Architecture
+
+* [Overview](#overview)
+    * [Hypervisor](#hypervisor)
+    * [Agent](#agent)
+    * [Runtime](#runtime)
+        * [create](#create)
+        * [start](#start)
+        * [exec](#exec)
+        * [kill](#kill)
+        * [delete](#delete)
+    * [Proxy](#proxy)
+    * [Shim](#shim)
+    * [Networking](#networking)
+* [Appendices](#appendices)
+    * [DAX](#dax)
+    * [Previous Releases](#previous-releases)
+        * [V2.0](#version-20)
+        * [V1.0](#version-10)
+
+## Overview
+
+This is an architectural overview of Clear Containers, based on the 3.0 release.
+
+The [Clear Containers runtime (cc-runtime)](https://github.com/clearcontainers/runtime)
+complies with the [OCI](https://github.com/opencontainers) specifications and thus
+works seamlessly with the [Docker Engine](https://www.docker.com/products/docker-engine)
+pluggable runtime architecture. In other words, one can transparently replace the
+[default Docker runtime (runc)](https://github.com/opencontainers/runc) with `cc-runtime`.
+
+![Runtime and virtcontainers](runtime-vc-relationship.png)
+![Docker and Clear Containers](docker-cc.png)
+
+`cc-runtime` creates a QEMU/KVM virtual machine for each container the Docker engine creates.
+
+The container process is then spawned by an [agent](https://github.com/clearcontainers/agent),
+running as a daemon on the guest operating system.
+Hyperstart opens 2 virtio serial interfaces (Control and I/O) on the guest, and QEMU exposes them
+as serial devices on the host. `cc-runtime` uses the control device for sending container
+management commands to the agent while the I/O serial device is used to pass I/O streams (`stdout`,
```

Comment:
Would be good to point to a section that contain the container management command or maybe the source file where the command protocol is defined.

Reply:
Added url to to godoc proxy api protocol, and added link to UML sequence diagram.